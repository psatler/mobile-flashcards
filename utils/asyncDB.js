import { AsyncStorage } from 'react-native'
import { data } from './mockData'
const FLASHCARDS_DECKS_KEY = 'Flashcards:decks'

const setInitialData = async () => {
    let obj = data;

    await AsyncStorage.setItem(FLASHCARDS_DECKS_KEY, JSON.stringify(obj))
}


export const getDecks = async () => {
    try {
        let value = await AsyncStorage.getItem(FLASHCARDS_DECKS_KEY);
        if( value !== null){ //we have data
            // console.log('decks', value);
            let parsedObj = JSON.parse(value); //parsing back to object
            return parsedObj;
        } else { //if not, we set initial state
            console.log('setting initial state');
            setInitialData(); //defined above
        }
    } catch (err) {
        console.log('There was an error retrieving decks from local DB: ', err);
    }
}

export const saveDeckTitle = (keyTitle, deckObj) => {
    try {
        return AsyncStorage.mergeItem(FLASHCARDS_DECKS_KEY, JSON.stringify({
            [keyTitle]: deckObj
        }))
        
    } catch (err) {
        console.log(err);
    }
}

export const addCardToDeck = (deckTitle, card) => {
    const key = deckTitle;

    return AsyncStorage.getItem(FLASHCARDS_DECKS_KEY)
        .then( (results) => {
            const allDecks = JSON.parse(results);
            const deck = allDecks[key];
            // console.log('addCardToDeck- BEFORE', deck.questions)
            deck.questions.push(card)
            // console.log('addCardToDeck', deck.questions)
            // console.log('allDecks', allDecks)

            AsyncStorage.setItem(FLASHCARDS_DECKS_KEY, JSON.stringify(allDecks)); //updating 

        })
        // const key = deckTitle.split(' ').join('');

}

// 

// const checkIfDataExists = (results) => {
//     return results === null
//       ? setInitialData()
//       : JSON.parse(results)
//   }

// export const retrieveDataFromAsyncStorage = () => {

//     return AsyncStorage.getItem(FLASHCARDS_DECKS_KEY)
//         .then(checkIfDataExists)
// }



// _setValue = async () => {
//     try {
//       await AsyncStorage.setItem(STORAGE_KEY, 'true');
//     } catch (error) { // log the error
//     }
//   };

//   _loadInitialState = async () => {
//     try {
//       let value = await AsyncStorage.getItem(STORAGE_KEY);
//       if (value === 'true'){
//         this.setState({loaded: 'true'});
//       } else {
//         this.setState({loaded: 'false'});
//         this._setValue();
//       }
//     } catch (error) {
//       this.setState({loaded: 'false'});
//       this._setValue();
//     }
//   };