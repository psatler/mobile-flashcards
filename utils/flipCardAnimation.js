import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  Animated
} from 'react-native';

export default class AnimatedBasic extends Component {

    state = {
        animatedValue: new Animated.Value(0),
        listenerValue: 0,
    }

    componentDidMount(){
        const { animatedValue, listenerValue } = this.state;

        //creating a listener to track values changes
        animatedValue.addListener( ( { value } ) => {
            this.setState({
                listenerValue: value,
            })
        })
    }

    flipCard = () => {
        const { animatedValue, listenerValue } = this.state;

        if(listenerValue >= 90){
            Animated.spring(animatedValue,{
                toValue: 0,
                friction: 8,
                tension: 10
              }).start();
        } 
        else {
            Animated.spring(animatedValue,{
                toValue: 180,
                friction: 8,
                tension: 10
              }).start();
        }
    }

    render() {
        const { animatedValue, listenerValue } = this.state;

        const frontAnimatedStyle = {
          transform: [
            { rotateY: animatedValue.interpolate({
                inputRange: [0, 180],
                outputRange: ['0deg', '180deg'],
              })
            }
          ],
          //adding opacity so the back face won't be mirrored on Android devices
        //https://github.com/facebook/react-native/issues/1973#issuecomment-262059217
          opacity: animatedValue.interpolate({ //adding this so it won't be mirrored on Android devices
                inputRange: [89, 90],
                outputRange: [1, 0]
            }), 
        }
        const backAnimatedStyle = {
          transform: [
            { rotateY: animatedValue.interpolate({
                inputRange: [0, 180],
              //   outputRange: ['180deg', '360deg'],
                outputRange: ['180deg', '0deg']
              }) 
            }
          ],
          opacity: animatedValue.interpolate({  //adding this so it won't be mirrored on Android devices
                inputRange: [89, 90], 
                outputRange: [0, 1] 
            }), 
        }
        
        return (
          <View style={styles.container}>
            <View>
              <Animated.View style={[styles.flipCard, frontAnimatedStyle]}>
                <Text style={styles.flipText}>
                  This text is flipping on the front.
                </Text>
              </Animated.View>
              <Animated.View style={[backAnimatedStyle, styles.flipCard, styles.flipCardBack]}>
                <Text style={styles.flipText}>
                  This text is flipping on the back.
                </Text>
              </Animated.View>
            </View>
            <TouchableOpacity onPress={() => this.flipCard()}>
              <Text>Flip!</Text>
            </TouchableOpacity>
          </View>
        );
      }
    }


const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  flipCard: {
    width: 200,
    height: 200,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'blue',
    backfaceVisibility: 'hidden', //it means when the card rotates out of the way, it won't be visible. When it rotates back, it'll be visible again. Not working on Android though, so it was added an workaround for this above
  },
  flipCardBack: {
    backgroundColor: "red",
    position: "absolute",
    top: 0,
  },
  flipText: {
    width: 90,
    fontSize: 20,
    color: 'white',
    fontWeight: 'bold',
  }
});