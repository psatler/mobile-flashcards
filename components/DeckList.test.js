import React from 'react'
import ConnectedDeckList, { DeckList } from './DeckList' 
import { data } from '../utils/mockData'


//####################################### ENZYME STUFF
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })
import { shallow, mount } from 'enzyme'

//####################################### REDUX MOCK STORE
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
const mockStore = configureMockStore([ thunk ]);
// const initialState = data['React'];
const initialState = {
    state: {
        deckReducer: {
            ...data,
        }
    }
}
const storeMock = mockStore(initialState);



describe('[Component] DeckList', () => {
    const navigation = { navigate: jest.fn() }; //as shown here https://stackoverflow.com/questions/46090713/testing-component-that-uses-react-navigation-with-jest/46114589

    it('shallow renders ConnectedDeckList correctly', () => {        
        const wrapper = shallow(
            // <ConnectedDeckList navigation={navigationMock} />,
            <ConnectedDeckList  />,
            { context: { store: storeMock } },
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });


    it('shallow renders Unconnected DeckList correctly', () => {        
        const wrapper = shallow(<DeckList />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders List of Decks correctly', () => {        
        const wrapper = shallow(<DeckList decks={data} />);
        wrapper.setState({ isLoading: true}) //setting a new state to test another part of the code
        expect(wrapper).toMatchSnapshot();
    });

    it('presses the touchableOpacity going to DeckDetail screen', () => {
        const wrapper = shallow(<DeckList decks={data} navigation={navigation}  />);
        wrapper.setState({ isLoading: true}) //
        wrapper.find('TouchableOpacity').first().props().onPress()
        expect(navigation.navigate).toHaveBeenCalledTimes(1);
    });

    it('calls the showDeleteConfirmation method', () => {
        const wrapper = shallow(<DeckList decks={data} navigation={navigation}  />);
        wrapper.setState({ isLoading: true}) //
        wrapper.find('TouchableOpacity').last().props().onPress() //this is the showDeleteConfirmation method (the last touchableOpacity)
        expect(navigation.navigate).toHaveBeenCalledTimes(1);
    });

    // it('calls getDecks method inside componentDidMount', () => {
    //     // console.log(DeckList.prototype)
    //     const spy = jest.spyOn(DeckList.prototype, 'getDecks');
    //     const wrapper = mount(<DeckList decks={data} navigation={navigation}  />);
    //     wrapper.instance().getDecks();
    //     expect(spy).toHaveBeenCalled();
    // })
});


