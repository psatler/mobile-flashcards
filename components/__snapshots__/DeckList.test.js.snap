// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] DeckList renders List of Decks correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeckList
    decks={
      Object {
        "JavaScript": Object {
          "questions": Array [
            Object {
              "answer": "The combination of a function and the lexical environment within which that function was declared.",
              "question": "What is a closure?",
            },
          ],
          "title": "JavaScript",
        },
        "React": Object {
          "questions": Array [
            Object {
              "answer": "A library for managing user interfaces",
              "question": "What is React?",
            },
            Object {
              "answer": "The componentDidMount lifecycle event",
              "question": "Where do you make Ajax requests in React?",
            },
          ],
          "title": "React",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ScrollView
          onScroll={[Function]}
          scrollEventThrottle={16}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Component
            style={
              Object {
                "marginTop": 120,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Deck
                deckName="React"
                deckSize={2}
                imageURI={undefined}
              />
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Deck
                deckName="JavaScript"
                deckSize={1}
                imageURI={undefined}
              />
            </TouchableOpacity>
          </Component>
        </ScrollView>,
        <AnimatedComponent
          style={
            Array [
              Object {
                "backgroundColor": "#00A4D3",
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 120,
              },
            ]
          }
        >
          <AnimatedComponent
            source={1}
            style={
              Array [
                Object {
                  "height": 120,
                  "left": 0,
                  "position": "absolute",
                  "resizeMode": "cover",
                  "right": 0,
                  "top": 0,
                  "width": null,
                },
                Object {
                  "opacity": 1,
                  "transform": Array [
                    Object {
                      "translateY": 0,
                    },
                  ],
                },
              ]
            }
          />
          <AnimatedComponent
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "height": 32,
                "justifyContent": "center",
                "left": 0,
                "marginTop": 28,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <AnimatedComponent
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#fff",
                    "fontSize": 18,
                  },
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 0,
                      },
                    ],
                  },
                ]
              }
            >
              Home
            </AnimatedComponent>
          </AnimatedComponent>
          <Component
            style={
              Object {
                "alignSelf": "flex-end",
                "right": 10,
                "top": 10,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Icon
                allowFontScaling={false}
                color="#fff"
                name="delete"
                size={30}
              />
            </TouchableOpacity>
          </Component>
        </AnimatedComponent>,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Component
            style={
              Object {
                "marginTop": 120,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Deck
                deckName="React"
                deckSize={2}
                imageURI={undefined}
              />
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
            >
              <Deck
                deckName="JavaScript"
                deckSize={1}
                imageURI={undefined}
              />
            </TouchableOpacity>
          </Component>,
          "onScroll": [Function],
          "scrollEventThrottle": 16,
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Deck
                  deckName="React"
                  deckSize={2}
                  imageURI={undefined}
                />
              </TouchableOpacity>,
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Deck
                  deckName="JavaScript"
                  deckSize={1}
                  imageURI={undefined}
                />
              </TouchableOpacity>,
            ],
            "style": Object {
              "marginTop": 120,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "React",
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.2,
                "children": <Deck
                  deckName="React"
                  deckSize={2}
                  imageURI={undefined}
                />,
                "onPress": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": "React",
                "nodeType": "function",
                "props": Object {
                  "deckName": "React",
                  "deckSize": 2,
                  "imageURI": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "JavaScript",
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.2,
                "children": <Deck
                  deckName="JavaScript"
                  deckSize={1}
                  imageURI={undefined}
                />,
                "onPress": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": "JavaScript",
                "nodeType": "function",
                "props": Object {
                  "deckName": "JavaScript",
                  "deckSize": 1,
                  "imageURI": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [MockFunction],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <AnimatedComponent
              source={1}
              style={
                Array [
                  Object {
                    "height": 120,
                    "left": 0,
                    "position": "absolute",
                    "resizeMode": "cover",
                    "right": 0,
                    "top": 0,
                    "width": null,
                  },
                  Object {
                    "opacity": 1,
                    "transform": Array [
                      Object {
                        "translateY": 0,
                      },
                    ],
                  },
                ]
              }
            />,
            <AnimatedComponent
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "height": 32,
                  "justifyContent": "center",
                  "left": 0,
                  "marginTop": 28,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <AnimatedComponent
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#fff",
                      "fontSize": 18,
                    },
                    Object {
                      "transform": Array [
                        Object {
                          "scale": 0,
                        },
                      ],
                    },
                  ]
                }
              >
                Home
              </AnimatedComponent>
            </AnimatedComponent>,
            <Component
              style={
                Object {
                  "alignSelf": "flex-end",
                  "right": 10,
                  "top": 10,
                }
              }
            >
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Icon
                  allowFontScaling={false}
                  color="#fff"
                  name="delete"
                  size={30}
                />
              </TouchableOpacity>
            </Component>,
          ],
          "style": Array [
            Object {
              "backgroundColor": "#00A4D3",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 120,
            },
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
              "style": Array [
                Object {
                  "height": 120,
                  "left": 0,
                  "position": "absolute",
                  "resizeMode": "cover",
                  "right": 0,
                  "top": 0,
                  "width": null,
                },
                Object {
                  "opacity": 1,
                  "transform": Array [
                    Object {
                      "translateY": 0,
                    },
                  ],
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <AnimatedComponent
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#fff",
                      "fontSize": 18,
                    },
                    Object {
                      "transform": Array [
                        Object {
                          "scale": 0,
                        },
                      ],
                    },
                  ]
                }
              >
                Home
              </AnimatedComponent>,
              "style": Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "height": 32,
                "justifyContent": "center",
                "left": 0,
                "marginTop": 28,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Home",
                "style": Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#fff",
                    "fontSize": 18,
                  },
                  Object {
                    "transform": Array [
                      Object {
                        "scale": 0,
                      },
                    ],
                  },
                ],
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Icon
                  allowFontScaling={false}
                  color="#fff"
                  name="delete"
                  size={30}
                />
              </TouchableOpacity>,
              "style": Object {
                "alignSelf": "flex-end",
                "right": 10,
                "top": 10,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.2,
                "children": <Icon
                  allowFontScaling={false}
                  color="#fff"
                  name="delete"
                  size={30}
                />,
                "onPress": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "allowFontScaling": false,
                  "color": "#fff",
                  "name": "delete",
                  "size": 30,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ScrollView
            onScroll={[Function]}
            scrollEventThrottle={16}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <Component
              style={
                Object {
                  "marginTop": 120,
                }
              }
            >
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Deck
                  deckName="React"
                  deckSize={2}
                  imageURI={undefined}
                />
              </TouchableOpacity>
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Deck
                  deckName="JavaScript"
                  deckSize={1}
                  imageURI={undefined}
                />
              </TouchableOpacity>
            </Component>
          </ScrollView>,
          <AnimatedComponent
            style={
              Array [
                Object {
                  "backgroundColor": "#00A4D3",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {
                  "height": 120,
                },
              ]
            }
          >
            <AnimatedComponent
              source={1}
              style={
                Array [
                  Object {
                    "height": 120,
                    "left": 0,
                    "position": "absolute",
                    "resizeMode": "cover",
                    "right": 0,
                    "top": 0,
                    "width": null,
                  },
                  Object {
                    "opacity": 1,
                    "transform": Array [
                      Object {
                        "translateY": 0,
                      },
                    ],
                  },
                ]
              }
            />
            <AnimatedComponent
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "height": 32,
                  "justifyContent": "center",
                  "left": 0,
                  "marginTop": 28,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <AnimatedComponent
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#fff",
                      "fontSize": 18,
                    },
                    Object {
                      "transform": Array [
                        Object {
                          "scale": 0,
                        },
                      ],
                    },
                  ]
                }
              >
                Home
              </AnimatedComponent>
            </AnimatedComponent>
            <Component
              style={
                Object {
                  "alignSelf": "flex-end",
                  "right": 10,
                  "top": 10,
                }
              }
            >
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Icon
                  allowFontScaling={false}
                  color="#fff"
                  name="delete"
                  size={30}
                />
              </TouchableOpacity>
            </Component>
          </AnimatedComponent>,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Component
              style={
                Object {
                  "marginTop": 120,
                }
              }
            >
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Deck
                  deckName="React"
                  deckSize={2}
                  imageURI={undefined}
                />
              </TouchableOpacity>
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
              >
                <Deck
                  deckName="JavaScript"
                  deckSize={1}
                  imageURI={undefined}
                />
              </TouchableOpacity>
            </Component>,
            "onScroll": [Function],
            "scrollEventThrottle": 16,
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <TouchableOpacity
                  activeOpacity={0.2}
                  onPress={[Function]}
                >
                  <Deck
                    deckName="React"
                    deckSize={2}
                    imageURI={undefined}
                  />
                </TouchableOpacity>,
                <TouchableOpacity
                  activeOpacity={0.2}
                  onPress={[Function]}
                >
                  <Deck
                    deckName="JavaScript"
                    deckSize={1}
                    imageURI={undefined}
                  />
                </TouchableOpacity>,
              ],
              "style": Object {
                "marginTop": 120,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "React",
                "nodeType": "class",
                "props": Object {
                  "activeOpacity": 0.2,
                  "children": <Deck
                    deckName="React"
                    deckSize={2}
                    imageURI={undefined}
                  />,
                  "onPress": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "React",
                  "nodeType": "function",
                  "props": Object {
                    "deckName": "React",
                    "deckSize": 2,
                    "imageURI": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "JavaScript",
                "nodeType": "class",
                "props": Object {
                  "activeOpacity": 0.2,
                  "children": <Deck
                    deckName="JavaScript"
                    deckSize={1}
                    imageURI={undefined}
                  />,
                  "onPress": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": "JavaScript",
                  "nodeType": "function",
                  "props": Object {
                    "deckName": "JavaScript",
                    "deckSize": 1,
                    "imageURI": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [MockFunction],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <AnimatedComponent
                source={1}
                style={
                  Array [
                    Object {
                      "height": 120,
                      "left": 0,
                      "position": "absolute",
                      "resizeMode": "cover",
                      "right": 0,
                      "top": 0,
                      "width": null,
                    },
                    Object {
                      "opacity": 1,
                      "transform": Array [
                        Object {
                          "translateY": 0,
                        },
                      ],
                    },
                  ]
                }
              />,
              <AnimatedComponent
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "height": 32,
                    "justifyContent": "center",
                    "left": 0,
                    "marginTop": 28,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <AnimatedComponent
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#fff",
                        "fontSize": 18,
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "scale": 0,
                          },
                        ],
                      },
                    ]
                  }
                >
                  Home
                </AnimatedComponent>
              </AnimatedComponent>,
              <Component
                style={
                  Object {
                    "alignSelf": "flex-end",
                    "right": 10,
                    "top": 10,
                  }
                }
              >
                <TouchableOpacity
                  activeOpacity={0.2}
                  onPress={[Function]}
                >
                  <Icon
                    allowFontScaling={false}
                    color="#fff"
                    name="delete"
                    size={30}
                  />
                </TouchableOpacity>
              </Component>,
            ],
            "style": Array [
              Object {
                "backgroundColor": "#00A4D3",
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": 120,
              },
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": 1,
                "style": Array [
                  Object {
                    "height": 120,
                    "left": 0,
                    "position": "absolute",
                    "resizeMode": "cover",
                    "right": 0,
                    "top": 0,
                    "width": null,
                  },
                  Object {
                    "opacity": 1,
                    "transform": Array [
                      Object {
                        "translateY": 0,
                      },
                    ],
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <AnimatedComponent
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#fff",
                        "fontSize": 18,
                      },
                      Object {
                        "transform": Array [
                          Object {
                            "scale": 0,
                          },
                        ],
                      },
                    ]
                  }
                >
                  Home
                </AnimatedComponent>,
                "style": Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "height": 32,
                  "justifyContent": "center",
                  "left": 0,
                  "marginTop": 28,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Home",
                  "style": Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#fff",
                      "fontSize": 18,
                    },
                    Object {
                      "transform": Array [
                        Object {
                          "scale": 0,
                        },
                      ],
                    },
                  ],
                },
                "ref": null,
                "rendered": "Home",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <TouchableOpacity
                  activeOpacity={0.2}
                  onPress={[Function]}
                >
                  <Icon
                    allowFontScaling={false}
                    color="#fff"
                    name="delete"
                    size={30}
                  />
                </TouchableOpacity>,
                "style": Object {
                  "alignSelf": "flex-end",
                  "right": 10,
                  "top": 10,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "activeOpacity": 0.2,
                  "children": <Icon
                    allowFontScaling={false}
                    color="#fff"
                    name="delete"
                    size={30}
                  />,
                  "onPress": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "allowFontScaling": false,
                    "color": "#fff",
                    "name": "delete",
                    "size": 30,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`[Component] DeckList shallow renders ConnectedDeckList correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeckList
    decks={undefined}
    dispatch={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;

exports[`[Component] DeckList shallow renders Unconnected DeckList correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeckList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
